core/src/main/scala/akka/kafka/internal/SingleSourceLogic.scala

```diff
@@ -100,6 +105,11 @@ private[kafka] abstract class SingleSourceLogic[K, V, Msg](
 
   }
 
+  private def invokeUserOnAssign(set: Set[TopicPartition]): Unit =
+    subscription.rebalanceListenerCallbacks.foreach(_.onAssign(set))
```

Comment:
do we need to protect ourselves more for exceptions thrown by user code?

Reply:
We do:

```
try invokeUserOnAssign(newTps)
      finally requestMessages()
```

Reply:
but it would still result in restart of the KafkaConsumerActor if exception thrown

Reply:
Hmm right; I can add a catch + log or move to the actor ref style, any preference?